apisx/v1alpha2/shared_types.go

```diff
@@ -0,0 +1,48 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha2
+
+import (
+	v1 "sigs.k8s.io/gateway-api/apis/v1"
+	v1alpha2 "sigs.k8s.io/gateway-api/apis/v1alpha2"
+)
+
+type (
+	// +k8s:deepcopy-gen=false
+	SessionPersistence = v1.SessionPersistence
+	// +k8s:deepcopy-gen=false
+	Duration = v1.Duration
+	// +k8s:deepcopy-gen=false
+	PolicyStatus = v1alpha2.PolicyStatus
+	// +k8s:deepcopy-gen=false
+	LocalPolicyTargetReference = v1alpha2.LocalPolicyTargetReference
+)
+
+// RequestRate expresses a rate of requests over a given period of time.
+type RequestRate struct {
+	// Count specifies the number of requests per time interval.
+	//
+	// Support: Extended
+	// +kubebuilder:validation:Minimum=0
+	Count *int `json:"count,omitempty"`
```

Comment:
We need to avoid unbounded values here, I'd recommend some kind of max here, even if that max is very high.

Reply:
The context in which this is currently used is for MinRetryRate as "an override for very low volume traffic" _but_ this struct type is (intentionally) sufficiently generic that it could be re-used in the future for https://github.com/kubernetes-sigs/gateway-api/issues/326. As such, I would defer to @htuch on what a "safe" maximum might entail for this at scale with a potential denominator of as short as `1ms`

Reply:
Given this is going to have a variable denominator, e.g. hour, I think you're just going to have to go with some type based limit or something that would make sense over a long period.